src/Responses/SaveResponse.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Strictphp\HttpClients\Responses;
+
+use Psr\Http\Message\ResponseInterface;
+use SplFileObject;
+use Strictphp\HttpClients\Contracts\FindExtensionFromHeadersActionContract;
+use Strictphp\HttpClients\Contracts\MakePathActionContract;
+use Strictphp\HttpClients\Entities\HttpStateEntity;
+use Strictphp\HttpClients\Events\SuccessRequestEvent;
+use Strictphp\HttpClients\Helpers\Headers;
+
+/**
+ * Use for event
+ * @see SuccessRequestEvent
+ */
+final class SaveResponse
+{
+    public function __construct(
+        private readonly MakePathActionContract $makePathAction,
+        private readonly FindExtensionFromHeadersActionContract $findExtensionFromHeaders,
+        private readonly int $bufferSize = 8192,
+    ) {
+    }
+
+    public function save(HttpStateEntity $httpState, ResponseInterface $response): void
```

Comment:
tady máme teda změnu, že už to neukládáme jako původní "soubor" (xml / json) ale jako vlastní soubor?

Reply:
je tu změna, že jsou hlavičky zvlášť a samotný soubor, kde se podle hlavičky typne koncovka souboru, aby ti OS přiřadil program